src/utils/googleTagManager.ts

```diff
@@ -155,14 +166,63 @@ export const trackEvent = ({
   action: string
   label?: EventLabel
 }): void => {
-  const dataLayer = {
+  const dataLayer: EventDataLayer = {
     event,
     chainId: _getChainId(),
     eventCategory: category,
     eventAction: action,
     eventLabel: tryParse(label),
   }
 
+  track(dataLayer)
+}
+
+const ETHEREUM_ADDRESS_REGEX = /0x[a-fA-F0-9]{40}/g
+const ETHEREUM_ADDRESS = 'ETHEREUM_ADDRESS'
+
+type SafeAppEventDataLayer = {
+  event: GTM_EVENT
+  chainId: string
+  safeAppName: string
+  safeAppMethod: string
+  safeAppParams?: string
+  safeAppEthMethod?: string
+  safeAppDeprecatedMethod: boolean
+  safeAppSDKVersion?: string
+}
+
+export const trackSafeAppEvent = ({
+  event,
+  app,
+  method,
+  params,
+  sdkVersion,
+  deprecated = false,
+}: {
+  event: GTM_EVENT
+  app?: SafeApp
+  method: string
+  params: any
+  sdkVersion?: string
+  deprecated?: boolean
+}): void => {
+  const safeApp = JSON.parse(app?.id || 'unknown')
+
+  const dataLayer: SafeAppEventDataLayer = {
+    event,
+    chainId: _getChainId(),
+    safeAppName: safeApp.name || safeApp.url || safeApp,
+    safeAppMethod: method,
+    safeAppParams: params ? JSON.stringify(params).replaceAll(ETHEREUM_ADDRESS_REGEX, ETHEREUM_ADDRESS) : undefined,
```

Comment:
You won't be able to parse this data in GA so it won't be analysable.

`replaceAll` isn't supported in all browsers. I'd suggest using a global `replace` instead.